manifest.yml

```diff
@@ -0,0 +1,105 @@
+    track: main
+    url: git@github.com:canonical/cos-lite-bundle.git
+
+  # charms
+  - path: ../alertmanager-k8s-operator
+    track: main
+    url: git@github.com:canonical/alertmanager-k8s-operator.git
+  
+  - path: ../avalanche-k8s-operator
+    track: main
+    url: git@github.com:canonical/avalanche-k8s-operator.git
+  
+  - path: ../catalogue-k8s-operator
+    track: main
+    url: git@github.com:canonical/catalogue-k8s-operator.git
+  
+  - path: ../cos-configuration-k8s-operator
+    track: main
+    url: git@github.com:canonical/cos-configuration-k8s-operator.git
+  
+  - path: ../cos-proxy-operator
+    track: main
+    url: git@github.com:canonical/cos-proxy-operator.git
+  
+  - path: ../cos-proxy-operator
+    track: main
+    url: git@github.com:canonical/cos-proxy-operator.git
+  
+  - path: ../grafana-agent-k8s-operator
+    track: main
+    url: git@github.com:canonical/grafana-agent-k8s-operator.git
+
+  - path: ../grafana-k8s-operator
+    track: main
+    url: git@github.com:canonical/grafana-k8s-operator.git
+  
+  - path: ../karma-alertmanager-proxy-k8s-operator
+    track: main
+    url: git@github.com:canonical/karma-alertmanager-proxy-k8s-operator.git
+  
+  - path: ../karma-k8s-operator
+    track: main
+    url: git@github.com:canonical/karma-k8s-operator.git
+  
+  - path: ../loki-k8s-operator
+    track: main
+    url: git@github.com:canonical/loki-k8s-operator.git
+  
+  - path: ../mimir-k8s-operator
+    track: main
+    url: git@github.com:canonical/mimir-k8s-operator.git
+  
+  - path: ../prometheus-k8s-operator
+    track: main
+    url: git@github.com:canonical/prometheus-k8s-operator.git
+  
+  - path: ../prometheus-pushgateway-k8s-operator
+    track: main
+    url: git@github.com:canonical/prometheus-pushgateway-k8s-operator.git
+
+  - path: ../prometheus-scrape-config-k8s-operator
+    track: main
+    url: git@github.com:canonical/prometheus-scrape-config-k8s-operator.git
+
+  - path: ../prometheus-scrape-target-k8s-operator
+    track: main
+    url: git@github.com:canonical/prometheus-scrape-target-k8s-operator.git
+  
+  - path: ../traefik-k8s-operator
+    track: main
+    url: git@github.com:canonical/traefik-k8s-operator.git
+  
+  - path: ../traefik-route-k8s-operator
+    track: main
+    url: git@github.com:canonical/traefik-route-k8s-operator.git
+
+  # rocks
+  - path: ../mimir-rock
+    track: main
+    url: git@github.com:canonical/mimir-rock.git
+  
+  - path: ../prometheus-rock
+    track: main
+    url: git@github.com:canonical/prometheus-rock.git
+
+  # aux
+  
+  - path: ../observability-libs
+    track: main
+    url: git@github.com:canonical/observability-libs.git
+  
+  - path: ../cos-tool 
+    track: main
+    url: git@github.com:canonical/cos-tool.git
+
+  - path: ../nrpe_exporter 
+    track: main
+    url: git@github.com:canonical/nrpe_exporter.git
+  
+  - path: ../charming-actions
+    track: main
+    url: git@github.com:canonical/charming-actions.git
```

Comment:
This one fails to sync (yet the status looks fine).

```
Traceback (most recent call last):
  File "/home/ubuntu/.local/bin/git-meta", line 8, in <module>
    sys.exit(cli())
  File "/home/ubuntu/.local/lib/python3.10/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ubuntu/.local/lib/python3.10/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/metarepo/cli_decorators.py", line 29, in wrapper
    return func(loaded_manifest, root_path, *args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/metarepo/commands/sync_cmd.py", line 92, in sync
    if not future.result():
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/metarepo/commands/sync_cmd.py", line 37, in do_sync_repo
    fetch_result = repo.fetch(repo_data.track)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/metarepo/vcs_git.py", line 105, in fetch
    fetch_result = self._repo.remote("origin").fetch(ref, progress=progress_cb)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/git/remote.py", line 1007, in fetch
    res = self._get_fetch_info_from_stderr(proc, progress, kill_after_timeout=kill_after_timeout)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/git/remote.py", line 848, in _get_fetch_info_from_stderr
    proc.wait(stderr=stderr_text)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/git/cmd.py", line 604, in wait
    raise GitCommandError(remove_password_if_present(self.args), status, errstr)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v -- origin main
  stderr: 'fatal: Could not read from remote repository.'
```

Reply:
Im unable to reproduce this locally. Same issue if you start from a clean slate?

Reply:
Here's a clean repro:
```bash
git clone https://github.com/canonical/observability
cd observability
git checkout feat/add-metarepo-manifest
python3 -m virtualenv venv
source venv/bin/activate
pip install metarepo
git meta sync
```

Reply:
Following your steps, I get this:

```bash
Synchronizing 24 repositories
  ✓ ../cos-lite-bundle (Checked out main)
  ✓ ../alertmanager-k8s-operator (Checked out main)
  ✓ ../avalanche-k8s-operator (Checked out main)
  ✓ ../catalogue-k8s-operator (Checked out main)
  ✓ ../cos-configuration-k8s-operator (Checked out main)
  ✓ ../cos-proxy-operator (Checked out main)
  ✓ ../grafana-agent-k8s-operator (Checked out main)
  ✓ ../grafana-k8s-operator (Checked out main)
  ✓ ../karma-alertmanager-proxy-k8s-operator (Checked out main)
  ✓ ../karma-k8s-operator (Checked out main)
  ✓ ../loki-k8s-operator (Checked out main)
  ✓ ../mimir-k8s-operator (Checked out main)
  ✓ ../prometheus-k8s-operator (Checked out main)
  ✓ ../prometheus-pushgateway-k8s-operator (Checked out main)
  ✓ ../prometheus-scrape-config-k8s-operator (Checked out main)
  ✓ ../prometheus-scrape-target-k8s-operator (Checked out main)
  ✓ ../traefik-k8s-operator (Checked out main)
  ✓ ../traefik-route-k8s-operator (Checked out main)
  ✓ ../mimir-rock (Checked out main)
  ✓ ../prometheus-rock (Checked out main)
  ✓ ../observability-libs (Checked out main)
  ✓ ../cos-tool (Checked out main)
  ✓ ../nrpe_exporter (Checked out main)
  ✓ ../charming-actions (Checked out main)
```

@jnsgruk, @lucabello, would you mind trying and let me know what results you get?